app/controllers/users_controller.rb

```diff
@@ -58,11 +58,19 @@ def change_password
   end
 
   def show
-    @repo_user = User.find_by username: params[:username]
-    @github_username = Octokit::Client.new(access_token: @repo_user.access_token).login
-    @repositories = @repo_user.repositories.where(make_id: nil).page params[:page]
-    @makes = @repo_user.repositories.where.not(make_id: nil).page params[:page]
-    @photos = photo_hash
+    if (params[:username] == @user.username)
+      @repo_user = User.find_by username: params[:username]
+      @github_username = Octokit::Client.new(access_token: @repo_user.access_token).login
+      @repositories = @repo_user.repositories.where(make_id: nil).page params[:page]
+      @makes = @repo_user.repositories.where.not(make_id: nil).page params[:page]
+      @photos = photo_hash
+    else
+      @repo_user = User.find_by username: params[:username]
+      @github_username = Octokit::Client.new(access_token: @repo_user.access_token).login
+      @repositories = @repo_user.repositories.where(share_type: "public").where(make_id: nil).page params[:page]
```

Comment:
If `@repositories` is the only variable difference here I would put just that in the if/else block and have everything else not repeated